wip-0010.md

```diff
@@ -0,0 +1,85 @@
+4. Bundle the recovery script into the `witnet/witnet-rust:1.1.3` and `witnet/witnet-rust:latest` docker images, and modify `migrations.sh` so the recovery script is run upon starting or restarting a container.
+
+Nodes that are not on the `A` chain SHOULD upgrade to `1.1.3` as soon as possible.
+
+Nodes that are already on the `A` chain SHOULDN'T upgrade to `1.1.3`, as upgrading would make no difference to them, and they would be reducing the set of available connections to peers that are already on the leading chain.
+
+Node operators can check whether a local node is on the `A` chain by running this one-liner:
+```sh
+docker exec witnet_node sh -c "witnet node blockchain --epoch 248839 --limit 2 2>&1 | grep -q '#248921 had digest 7556670d' && echo 'Your node seems to be OK' || echo 'Oh no! Your node is forked'"
+```
+
+This is the equivalent for nodes not using docker:
+```sh
+./witnet node blockchain --epoch 248839 --limit 2 2>&1 | grep -q '#248921 had digest 7556670d' && echo 'Your node seems to be OK' || echo 'Oh no! Your node is forked'
+```
+
+### Medium term
+
+Even if the need for these *medium term* actions is truly pressing, they have been delayed because they will require a coordinated upgrade of every single node in the network, which cannot be trivially carried out without an advance notice period of several days or weeks.
+
+With regards to preventing situations like this from happening in the future, it is advised that the community starts working urgently on a new Witnet Improvement Proposal that introduces a floor to the superblock signing committee size.
+
+May that proposal gather consensus in the community in a short time, it would be advisable that it gets adopted along [WIP-0009] in `witnet-rust` version `1.2.0`. The activation date for [WIP-0009] could be postponed from the original date (March 16, 2021) if necessary.
+
+### Long term
+
+Beyond setting a minimum superblock committee size, these are some other ideas relevant to this issue that have suggested by different community members and are probably worth being further explored and discussed sooner or later:
+
+- Existence of a reliable "rescue committee" that can kick in and have "casting vote" rights when superblock consensus is lost. Identities could be added or removed from this committee through WIPs.
+- Incentivization for single superblock voting, either in form of rewards, penalties, or both.
+- Incentivization for nodes that accept inbound connections, so as to improve the ratio between public and private nodes.
+- A more convenient mechanism for automatically importing lists of peer addresses.
+- Some method for exporting and importing chain state snapshots, as a way to instantly getting the local chain into a particular block height.
```

Comment:
```suggestion
- Some method for exporting and importing chain state snapshots, as a way to instantly get the local chain into a particular block height.
```

Reply:
I'm a maximalist of this rule of thumb:
- to + infinitive
- of + gerund
- for + gerund

With the only exception of a few phrasal verbs, e.g. "look forward to", "be used to", "be committed to", "be devoted to", etc.